blocks/api/parser.js

```diff
@@ -96,57 +124,11 @@ export function asType( value, type ) {
  * @return {Object}                All block attributes
  */
 export function getBlockAttributes( blockType, rawContent, attributes ) {
-	// Retrieve additional attributes sourced from content
-	const sourcedAttributes = getSourcedAttributes(
-		rawContent,
-		blockType.attributes
-	);
-
-	const blockAttributes = reduce( blockType.attributes, ( result, source, key ) => {
-		let value;
-		if ( sourcedAttributes.hasOwnProperty( key ) ) {
-			value = sourcedAttributes[ key ];
-		} else if ( attributes ) {
-			value = attributes[ key ];
-		}
-
-		// Return default if attribute value not assigned
-		if ( undefined === value ) {
-			// Nest the condition so that constructor coercion never occurs if
-			// value is undefined and block type doesn't specify default value
-			if ( 'default' in source ) {
-				value = source.default;
-			} else {
-				return result;
-			}
-		}
-
-		// Coerce value to specified type
-		const coercedValue = asType( value, source.type );
-
-		if ( 'development' === process.env.NODE_ENV &&
-				! sourcedAttributes.hasOwnProperty( key ) &&
-				value !== coercedValue ) {
-			// Only in case of sourcing attribute from content do we want to
-			// allow coercion, as comment attributes are serialized respecting
-			// original data type. In development environments, log if value
-			// coerced to specified type is not strictly equal. We still allow
-			// coerced value to be assigned into attributes to avoid errors.
-			//
-			// Example:
-			//   Number( 5 ) === 5
-			//   Number( '5' ) !== '5'
-
-			// eslint-disable-next-line no-console
-			console.error(
-				`Expected attribute "${ key }" of type ${ source.type } for ` +
-				`block type "${ blockType.name }" but received ${ typeof value }.`
-			);
-		}
-
-		result[ key ] = coercedValue;
-		return result;
-	}, {} );
+	const blockAttributes = keys( blockType.attributes ).reduce( ( memo, attributeKey ) => {
+		const attributeSchema = blockType.attributes[ attributeKey ];
+		memo[ attributeKey ] = getBlockAttribute( attributeKey, attributeSchema, rawContent, attributes );
+		return memo;
+	}, {} ) || {};
```

Comment:
I don't think the `|| {}` would ever be triggered? Even if the `reduce` operates on an empty set, the initial value should be returned.

```
> [].reduce( () => {}, {} );
{}
```